samples/ck-rx65n/basic-sample/src/main.c

```diff
@@ -0,0 +1,424 @@
+/**************************************************************************/
+
+#include <stdbool.h>
+#include "nx_api.h"
+#undef SAMPLE_DHCP_DISABLE
+#ifndef SAMPLE_DHCP_DISABLE
+#include "nxd_dhcp_client.h"
+#endif /* SAMPLE_DHCP_DISABLE */
+#include "nxd_dns.h"
+#include "nxd_sntp_client.h"
+#include "nx_secure_tls_api.h"
+#include "azrtos_time.h"
+#include "nx_driver_rx_fit.h"
+
+#include "demo_printf.h"
+#include "app_config.h"
+
+#include "hardware_setup.h"
+
+extern bool app_startup(NX_IP *ip_ptr, NX_PACKET_POOL *pool_ptr, NX_DNS *dns_ptr);
+
+/* Define the helper thread for running Azure SDK on ThreadX (THREADX IoT Platform).  */
+#ifndef SAMPLE_HELPER_STACK_SIZE
+#define SAMPLE_HELPER_STACK_SIZE      (7 * 1024)
+#endif /* SAMPLE_HELPER_STACK_SIZE  */
+
+#ifndef SAMPLE_HELPER_THREAD_PRIORITY
+#define SAMPLE_HELPER_THREAD_PRIORITY (4)
+#endif /* SAMPLE_HELPER_THREAD_PRIORITY  */
+
+/* Define user configurable symbols. */
+#ifndef SAMPLE_IP_STACK_SIZE
+#define SAMPLE_IP_STACK_SIZE          (2048)
+#endif /* SAMPLE_IP_STACK_SIZE  */
+
+#ifndef SAMPLE_PACKET_COUNT
+#define SAMPLE_PACKET_COUNT           (60)
+#endif /* SAMPLE_PACKET_COUNT  */
+
+#ifndef SAMPLE_PACKET_SIZE
+#define SAMPLE_PACKET_SIZE            (1536)
+#endif /* SAMPLE_PACKET_SIZE  */
+
+#define SAMPLE_POOL_SIZE              ((SAMPLE_PACKET_SIZE + sizeof(NX_PACKET)) * SAMPLE_PACKET_COUNT)
+
+#ifndef SAMPLE_ARP_CACHE_SIZE
+#define SAMPLE_ARP_CACHE_SIZE         (512)
+#endif /* SAMPLE_ARP_CACHE_SIZE  */
+
+#ifndef SAMPLE_IP_THREAD_PRIORITY
+#define SAMPLE_IP_THREAD_PRIORITY     (1)
+#endif /* SAMPLE_IP_THREAD_PRIORITY */
+
+#if __STDC_VERSION__ >= 201112UL
+
+#endif
+
+#if defined(SAMPLE_DHCP_DISABLE) && !defined(SAMPLE_NETWORK_CONFIGURE)
+#ifndef SAMPLE_IPV4_ADDRESS
+/*#define SAMPLE_IPV4_ADDRESS           IP_ADDRESS(192, 168, 100, 33)*/
+#error "SYMBOL SAMPLE_IPV4_ADDRESS must be defined. This symbol specifies the IP address of device. "
+#endif /* SAMPLE_IPV4_ADDRESS */
+
+#ifndef SAMPLE_IPV4_MASK
+/*#define SAMPLE_IPV4_MASK              0xFFFFFF00UL*/
+#error "SYMBOL SAMPLE_IPV4_MASK must be defined. This symbol specifies the IP address mask of device. "
+#endif /* SAMPLE_IPV4_MASK */
+
+#ifndef SAMPLE_GATEWAY_ADDRESS
+/*#define SAMPLE_GATEWAY_ADDRESS        IP_ADDRESS(192, 168, 100, 1)*/
+#error "SYMBOL SAMPLE_GATEWAY_ADDRESS must be defined. This symbol specifies the gateway address for routing. "
+#endif /* SAMPLE_GATEWAY_ADDRESS */
+
+#ifndef SAMPLE_DNS_SERVER_ADDRESS
+/*#define SAMPLE_DNS_SERVER_ADDRESS     IP_ADDRESS(192, 168, 100, 1)*/
+#error "SYMBOL SAMPLE_DNS_SERVER_ADDRESS must be defined. This symbol specifies the dns server address for routing. "
+#endif /* SAMPLE_DNS_SERVER_ADDRESS */
+#else
+#define SAMPLE_IPV4_ADDRESS             IP_ADDRESS(0, 0, 0, 0)
+#define SAMPLE_IPV4_MASK                IP_ADDRESS(0, 0, 0, 0)
+
+//#define SAMPLE_DNS_SERVER_ADDRESS       IP_ADDRESS(8, 8, 8, 8)
+#define SAMPLE_DNS_SERVER_ADDRESS       IP_ADDRESS(192, 168, 1, 1)

```

Comment:
Looks odd? Does everyone have a DNS setup at 192.168.1.1

Reply:
I think that in most other projects even though this is defined, it's not used because DNS server is obtained from DHCP negotiation. Those are maybe some sort of defaults to fallback id DHCP fails? You can maybe verify that in the log output. It should print the obtained DNS address from DHCP.